tests/test_quantum_gradients.py

```diff
@@ -607,3 +607,52 @@ def circuit(x):
                            match="Circuits that include sampling can not be differentiated."):
             grad_fn = autograd.jacobian(circuit)
             grad_fn(1.0)
+
+
+@pytest.mark.usefixtures("tape_mode")
```

Comment:
This test class is run in both tape *and* non-tape mode.

Reply:
How come similar cases were added separately to `tests/tape/tapes/test_qubit_param_shift.py`? :thinking:

Reply:
`tests/tape/tapes/test_qubit_param_shift.py` are tape specific tests, so all tests are done at the tape level. In this file, we are instead doing quantum gradient functional tests at the QNode level.

I thought for completeness, better to test all cases!